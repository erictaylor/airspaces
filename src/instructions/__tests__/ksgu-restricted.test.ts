import { describe, expect, it } from 'bun:test';
import {
  BEARING,
  SIDE,
  getEastExtensionInstructions,
  getPrimaryInstructions,
  getWestExtensionInstructions,
} from '../ksgu-restricted';

describe('KSGU Ultralight Restricted Instructions', () => {
  it('should return the expected primary instructions', () => {
    expect(getPrimaryInstructions(BEARING, SIDE)).toMatchSnapshot();
  });

  it('should return the expected east extension instructions', () => {
    expect(getEastExtensionInstructions(BEARING, SIDE)).toMatchSnapshot();
  });

  it('should return the expected west extension instructions', () => {
    expect(getWestExtensionInstructions(BEARING, SIDE)).toMatchSnapshot();
  });
});
