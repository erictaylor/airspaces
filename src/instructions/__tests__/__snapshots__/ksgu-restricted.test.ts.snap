// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`KSGU Ultralight Restricted Instructions should return the expected primary instructions 1`] = `
[
  "AC R",
  "AN KSGU ULTRALIGHT VEHICLES RESTRICTED PRIMARY",
  "AH 17999 ft",
  "AL SFC",
  "DP 037:05:27.08 N 113:29:49.20 W",
  "DP 037:04:50.54 N 113:28:06.32 W",
  "DP 036:59:35.93 N 113:31:01.88 W",
  "DP 037:00:12.47 N 113:32:44.64 W",
]
`;

exports[`KSGU Ultralight Restricted Instructions should return the expected east extension instructions 1`] = `
[
  "AC R",
  "AN KSGU ULTRALIGHT VEHICLES RESTRICTED EAST EXT",
  "AH 17999 ft",
  "AL SFC",
  "DP 037:03:03.49 N 113:29:06.05 W",
  "DP 037:01:23.80 N 113:29:07.50 W",
  "DP 037:00:37.40 N 113:29:59.50 W",
  "DP 037:00:34.42 N 113:30:29.23 W",
]
`;

exports[`KSGU Ultralight Restricted Instructions should return the expected west extension instructions 1`] = `
[
  "AC R",
  "AN KSGU ULTRALIGHT VEHICLES RESTRICTED WEST EXT",
  "AH 17999 ft",
  "AL SFC",
  "DP 037:02:29.27 N 113:31:28.41 W",
  "DP 037:00:12.47 N 113:32:44.64 W",
  "V X=037:02:29.27 N 113:31:28.41 W",
  "DB 037:00:12.47 N 113:32:44.64 W, 037:00:32.89 N 113:33:26.41 W",
  "V X=037:00:55.44 N 113:33:02.44 W",
  "DB 037:00:32.89 N 113:33:26.41 W, 037:01:10.89 N 113:33:34.02 W",
  "DP 037:02:29.27 N 113:31:28.41 W",
]
`;
